runtime-tests/tests/member_registration.rs

```diff
@@ -0,0 +1,239 @@
+    let org_id = random_id();
+    let register_org = message::RegisterOrg {
+        org_id: org_id.clone(),
+    };
+    submit_ok(&client, &good_actor, register_org.clone()).await;
+
+    // The bad actor attempts to register themselves as a member within that org.
+    let initial_balance = client.free_balance(&bad_actor.public()).await.unwrap();
+    let register_member = message::RegisterMember {
+        org_id,
+        user_id: bad_actor_id,
+    };
+    let random_fee = random_balance();
+    let tx_applied =
+        submit_ok_with_fee(&client, &bad_actor, register_member.clone(), random_fee).await;
+
+    assert_eq!(
+        tx_applied.result,
+        Err(RegistryError::InsufficientSenderPermissions.into())
+    );
+
+    // Check that the bad actor payed for the transaction anyway.
+    assert_eq!(
+        client.free_balance(&bad_actor.public()).await.unwrap(),
+        initial_balance - random_fee,
+        "The tx fee was not charged properly."
+    );
+
+    // Re-fetch the org and check that the bad actor was not added as a member
+    let re_org = client
+        .get_org(register_member.clone().org_id)
+        .await
+        .unwrap()
+        .unwrap();
+
+    assert_eq!(re_org.members, vec![good_actor_id]);
+}
+
+#[async_std::test]
+async fn register_duplicate_member() {
+    let client = Client::new_emulator();
+    let (author, author_id) = key_pair_with_associated_user(&client).await;
+
+    // Register the org.
+    let org_id = random_id();
+    let register_org = message::RegisterOrg {
+        org_id: org_id.clone(),
+    };
+    submit_ok(&client, &author, register_org.clone()).await;
+
+    // The org needs funds to submit transactions.
+    let org = client
+        .get_org(register_org.org_id.clone())
+        .await
+        .unwrap()
+        .unwrap();
+    let initial_balance = 1000;
+    transfer(&client, &author, org.account_id, initial_balance).await;
+
+    // The author attempts to register themselves as a member within that org again.
+    let register_member = message::RegisterMember {
+        org_id,
+        user_id: author_id.clone(),
+    };
+    let random_fee = random_balance();
+    let tx_applied =
+        submit_ok_with_fee(&client, &author, register_member.clone(), random_fee).await;
+
+    assert_eq!(
+        tx_applied.result,
+        Err(RegistryError::DuplicateMember.into())
+    );
+
+    // Re-fetch the org
+    let re_org = client
+        .get_org(register_member.clone().org_id)
+        .await
+        .unwrap()
+        .unwrap();
+
+    // Check that the org payed for the transaction anyway.
+    assert_eq!(
+        client.free_balance(&re_org.account_id).await.unwrap(),
+        initial_balance - random_fee,
+        "The tx fee was not charged properly."
+    );
+
+    // Check that the author was not added again
+    assert_eq!(re_org.members, vec![author_id]);
+}
+
+#[async_std::test]
+async fn register_nonexistent_member() {
```

Comment:
I got confused with the name for a sec, maybe:

```suggestion
async fn register_nonexistent_user() {
```